### ISSUE TITLE
ResNet50 from cifar10_resnet.py has lower test accuracy than elsewhere

### ISSUE THREAD

**Issue by ozabluda** [keras-team/keras#9452]
There is a (possibly erroneous, but credible) claim by @yaroslavvb

https://dawn.cs.stanford.edu/benchmark/#cifar10-train-time
https://github.com/ppwwyyxx/tensorpack/blob/master/examples/ResNet/cifar10-resnet.py

that his ResNet50 (ResNet v1 n=5? ) reaches >94% test accuracy on CIFAR-10, which is not happening with `cifar10_resnet.py` example
https://github.com/keras-team/keras/blob/master/examples/cifar10_resnet.py#L36

The difference is statistically significant.

**Comment by yaroslavvb** [comment]
This is a totally not-erroneous claim by me. I first tried official TensorFlow examples and found them to be not as tuned/optimized as @ppwwyyxx 's TensorPack models. One example issue is here https://github.com/tensorflow/models/issues/3311 . You can reproduce 94% accuracy by doing something like this

```
wget https://raw.githubusercontent.com/ppwwyyxx/tensorpack/master/examples/ResNet/cifar10-resnet.py
pip install tensorpack
python cifar10-resnet.py
```

**Comment by fchollet** [comment]
@roatienza FYI.

**Comment by ozabluda** [comment]
For the reference: https://github.com/keras-team/keras/pull/8231#issuecomment-353196678
>The following is a catalog of differences between the current cifar10_resnet.py example and the orig paper. I tried "fixing" some but not all of them, but didn't get statistically significant improvements, doesn't mean it's impossible.

**Comment by qichenglao** [comment]
I also tried ResNet20 from cifar10_resnet.py for at least 4 times, and didn't get comparable accuracy as stated in the table. (resnet20_v1: 89.88%, 90.46%, 89.80% and resnet_v2: 90.47%)

I am using Keras=2.1.4 and tensorflow-gpu=1.5.0

**Comment by roatienza** [comment]
On 1080Ti, first run achieves 92.16% for ResNet20 V1. keras 2.1.5 tf 1.6. Consistent with the table.

Epoch 00159: val_acc did not improve
Epoch 160/200
Learning rate:  1e-05
1563/1563 [==============================] - 26s 17ms/step - loss: 0.1761 - acc: 0.9824 - val_loss: 0.4274 - val_acc: 0.9216

Epoch 00160: val_acc improved from 0.92100 to 0.92160, saving model to /home/rowel/github/keras/examples/saved_models/cifar10_ResNet20v1_model.160.h5
Epoch 161/200
Learning rate:  1e-05
1563/1563 [==============================] - 26s 17ms/step - loss: 0.1735 - acc: 0.9837 - val_loss: 0.4256 - val_acc: 0.9215

**Comment by qichenglao** [comment]
@roatienza does GPU matter? I am using 1080.

It looks like I have slower convergence. The last epoch still gives a loss of 0.2283:

1563/1563 [==============================] - 49s 31ms/step - loss: 0.2283 - acc: 0.9669 - val_loss: 0.5080 - val_acc: 0.8988

Epoch 00200: val_acc did not improve

**Comment by roatienza** [comment]
@qichenglao Are you using tensorflow-gpu? Your time per epoch is ~2x than that of 1080Ti.

**Comment by qichenglao** [comment]
@roatienza right, it's quite slow especially with small batch size, but should not affect the performance, right?

**Comment by roatienza** [comment]
@qichenglao I normally notice differences in network performance between gpu and cpu but not between gpus
